internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -234,9 +234,9 @@ func (r *NumaflowControllerReconciler) reconcile(
 		numaLogger.Info("Deleting NumaflowController")
 		r.recorder.Eventf(controller, corev1.EventTypeNormal, "Deleting", "Deleting NumaflowController")
 		if controllerutil.ContainsFinalizer(controller, common.FinalizerName) {
-			// Check if dependent resources are deleted, if not then requeue after 5 seconds
+			// Check if dependent resources are deleted, if not then requeue
 			if !r.areDependentResourcesDeleted(ctx, controller) {
-				numaLogger.Warn("Dependent resources are not deleted yet, requeue after 5 seconds")
+				numaLogger.Warn("Dependent resources are not deleted yet, requeue...")
 				return ctrl.Result{Requeue: true}, nil
```

Comment:
So, I guess we aren't doing foreground deletion for this one - I guess that's okay since it seems like theoretically there shouldn't be anything holding up the deletion of those underlying children, but then again I wonder if we just want it to be consistent. Any reason you can think of not to just add it here too?

Reply:
I didn't added it here, because `NumaflowController_rollout` already has it and in `NumaflowController_controller` has this `areDependentResourcesDeleted` check

Reply:
right, so I guess the one reason to add it here would be if say there were some resource being deployed by the NumaflowController - I know right now it's just `ConfigMap`, `Deployment`, etc - but what if one of them had some finalizer that couldn't be deleted for some reason? At least, we'd be consistent everywhere in Numaplane with that behavior. I'm kind of leaning towards adding it, unless you disagree.

Reply:
Sure, I'll add it, I don't have strong reason against it, also it will make it consistent.